src/format/strftime.rs

```diff
@@ -169,18 +174,93 @@ pub struct StrftimeItems<'a> {
     /// If the current specifier is composed of multiple formatting items (e.g. `%+`),
     /// parser refers to the statically reconstructed slice of them.
     /// If `recons` is not empty they have to be returned earlier than the `remainder`.
+    #[cfg(feature = "locales")]
+    recons: Vec<Item<'a>>,
+    #[cfg(not(feature = "locales"))]
     recons: &'static [Item<'static>],
+    /// Date format
+    #[cfg(feature = "locales")]
+    d_fmt: Vec<Item<'a>>,
+    #[cfg(not(feature = "locales"))]
+    d_fmt: &'static [Item<'static>],
+    /// Date and time format
+    #[cfg(feature = "locales")]
+    d_t_fmt: Vec<Item<'a>>,
+    #[cfg(not(feature = "locales"))]
+    d_t_fmt: &'static [Item<'static>],
+    /// Time format
+    #[cfg(feature = "locales")]
+    t_fmt: Vec<Item<'a>>,
+    #[cfg(not(feature = "locales"))]
+    t_fmt: &'static [Item<'static>],
 }
 
 impl<'a> StrftimeItems<'a> {
     /// Creates a new parsing iterator from the `strftime`-like format string.
     pub fn new(s: &'a str) -> StrftimeItems<'a> {
-        static FMT_NONE: [Item<'static>; 0] = [];
-        StrftimeItems { remainder: s, recons: &FMT_NONE }
+        static D_FMT: &'static [Item<'static>] =
+            &[num0!(Month), lit!("/"), num0!(Day), lit!("/"), num0!(YearMod100)];
+        static D_T_FMT: &'static [Item<'static>] = &[
+            fix!(ShortWeekdayName),
+            sp!(" "),
+            fix!(ShortMonthName),
+            sp!(" "),
+            nums!(Day),
+            sp!(" "),
+            num0!(Hour),
+            lit!(":"),
+            num0!(Minute),
+            lit!(":"),
+            num0!(Second),
+            sp!(" "),
+            num0!(Year),
+        ];
+        static T_FMT: &'static [Item<'static>] =
+            &[num0!(Hour), lit!(":"), num0!(Minute), lit!(":"), num0!(Second)];
+        #[cfg(not(feature = "locales"))]
+        static FMT_NONE: &'static [Item<'static>; 0] = &[];
+
+        #[cfg(feature = "locales")]
+        {
+            StrftimeItems {
+                remainder: s,
+                recons: Vec::new(),
+                d_fmt: D_FMT.to_vec(),
+                d_t_fmt: D_T_FMT.to_vec(),
+                t_fmt: T_FMT.to_vec(),
+            }
+        }
```

Comment:
I prefer `cfgs` to be out at the function level, could you make a pair of cfg-different private `StrftimeItems::with_remainder` functions?

Reply:
done!

I moved the common statics to the module level to avoid duplication.